RileyLinkKitTests/PumpOpsSynchronousTests.swift

```diff
@@ -73,202 +84,289 @@ class PumpOpsSynchronousTests: XCTestCase {
+        assertArray(timestampedEvents, containsPumpEvent: event)
+        XCTAssertTrue(hasMoreEvents)
+        XCTAssertFalse(cancelled)
+    }
+
+    // MARK: Regular Bolus Event before starttime (offset 9 minutes)
+    // fixme: event is actually *after* start time, since startTime is adjusted to be 9 minutes before event
+    func test522RegularBolusEventBeforeStartTimeShouldNotCancel() {
+        setUpTestWithPumpModel(.Model522)
+        let event = createBolusEvent2009()
         
-        assertArray(timestampedEvents, containsPumpEvent: event2010)
+        let (_, _, cancelled) = runDeltaAllowanceTimeTest(pumpEvent: event, timeIntervalAdjustment: TimeInterval(minutes:-9))
+
+        XCTAssertFalse(cancelled)
     }
-    
+
+
+    // Remove or fixme: same idea as testEventsAfterStartDateAreReturned
+    func testShouldContainEventWhen522EstimatedTimeDeltaAllowanceBeforeAdjustedStartTime() {
+        setUpTestWithPumpModel(.Model522)
+
+        let bolusEvent = createBolusEvent2009()
+        let (timestampedEvents, _, _) = runDeltaAllowanceTimeTest(pumpEvent: bolusEvent, timeIntervalAdjustment: -1)
+        
+        assertArray(timestampedEvents, containsPumpEvent: bolusEvent)
+    }
+
+    // A normal bolus is skipped in the check for completion because it is a delayed append.  This reason for this behavior is tested
+    // in testDelayedAppendEventDoesNotCauseValidEventsToBeFilteredOut.  Recommend removing this
+//    func testShouldContain522EventWhenEstimateTimeDeltaAllowanceAfterAdjustedStartTime() {
+//        setUpTestWithPumpModel(.Model522)
+//
+//        let bolusEvent = createBolusEvent2009()
+//        let (timestampedEvents, hasMoreEvents, cancelled) = runDeltaAllowanceTimeTest(pumpEvent: bolusEvent, timeIntervalAdjustment: TimeInterval(hours:11))
+//
+//        assertArray(timestampedEvents, containsPumpEvent: bolusEvent)
+//        XCTAssertTrue(hasMoreEvents)
+//        XCTAssertFalse(cancelled)
+//    }
+
+    // A bolus on 523 is not delayed append, so we do filter it out if it is before startTime, just like any other event,
+    // which is tested in testEventBeforeStartDateIsFiltered.  I think this is redundant
+//    func testShouldNotCancelWhen523EstimateTimeDeltaAllowanceAfterAdjustedStartTime() {
+//        setUpTestWithPumpModel(.Model523)
+//
+//        let squareBolusEvent = createSquareBolusEvent2010()
+//        let (timestampedEvents, hasMoreEvents, cancelledEarly) = runDeltaAllowanceTimeTest(pumpEvent: squareBolusEvent, timeIntervalAdjustment: TimeInterval(hours:11))
+//
+//        assertArray(timestampedEvents, doesntContainPumpEvent: squareBolusEvent)
+//        XCTAssertFalse(cancelledEarly)
+//        XCTAssertFalse(hasMoreEvents)
+//    }
+
+
+    // The following tests are testing the border conditionsof the pump lost time detection, the main behavior of which is covered
```

Comment:
I'd feel better if we either kept the tests or removed the code they're testing. I don't like leaving in ambiguous code without tests. There may be functionality relying on the current implementation that we're not considering. This type of code can be easily removed by mistake while changing other functionality. 

I'd propose keeping one of these test, probably the one at 59 minutes. One benefit is so we know if the behavior of this method changes.

I'll let you make the call on this. Let me know.